src/scheduler.rs

```diff
@@ -0,0 +1,663 @@
+//! An unbounded set of futures.
+
+use Async;
+use executor::{self, UnsafeNotify, NotifyHandle};
+use task_impl::{AtomicTask, ThreadNotify};
+
+use std::cell::UnsafeCell;
+use std::fmt::{self, Debug};
+use std::marker::PhantomData;
+use std::mem;
+use std::ptr;
+use std::sync::atomic::Ordering::{Relaxed, SeqCst, Acquire, Release, AcqRel};
+use std::sync::atomic::{AtomicPtr, AtomicBool};
+use std::sync::{Arc, Weak};
+use std::usize;
+
+/// A generic task-aware scheduler.
```

Comment:
Could you add a more in-depth comment? Something to the effect of: Scheduler is used to run many child tasks inside of another thing that can be woken up via a "Wakeup".

Reply:
I would definitely appreciate a patch that improves on the docs 👍